internal/service/labelstore/service.go

```diff
@@ -155,6 +168,10 @@ func (s *service) GetOrAddLink(componentID string, localRefID uint64, lbls label
 
 // GetOrAddGlobalRefID is used to create a global refid for a labelset
 func (s *service) GetOrAddGlobalRefID(l labels.Labels) uint64 {
+	if !s.enabled.Load() {
+		return l.Hash()
```

Comment:
Should we guard against bad input like line 161?

Reply:
Probably doesnt hurt, will add.